app/repository/pg/user.go

```diff
@@ -27,10 +27,10 @@ func NewRepo(db *sql.DB) *Repo {
 // Add meth implements usecase.UserRepository interface
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
-func (r *Repo) Add(user entities.User) error {
+func (r Repo) Add(user *entities.User) error {
 	const sqlStatement = `INSERT INTO "user" (first_name, last_name, email, password)
-						  VALUES ($1, $2, $3, $4)`
-	if _, err := r.DB.Exec(sqlStatement, user.FirstName, user.LastName, user.Email, hash.SHA256(user.Password)); err != nil {
+						  VALUES ($1, $2, $3, $4) RETURNING id`
+	if err := r.DB.QueryRow(sqlStatement, user.FirstName, user.LastName, user.Email, hash.SHA256(user.Password)).Scan(&user.ID); err != nil {
```

Comment:
What is going on here? 😄

Reply:
@vpakhuchyi  I will try to explain. In the previous version of the code, we simply added the user to the database and at the same time logged all his fields, but the ID was empty because the ID is created in the database. And it seemed normal. Now we have decided that almost all user fields are sensitive data except ID (which is empty). There is no point in logging an empty value. So I changed the code so that the user ID is returned from the database after adding

Reply:
Thx @vpakhuchyi . I understood why there is no need to use a pointer here (because the signature of the function does not say that there will be any changes in the arguments sent). Therefore, I removed the pointers and changed the signature of the functions, which will clearly indicate that you need to expect an ID and an error